app/src/main/java/org/transitclock/core/reports/SqlUtils.java

```diff
@@ -217,10 +217,12 @@ public static String timeRangeClause(
 
         SimpleDateFormat currentFormat = new SimpleDateFormat("MM-dd-yyyy");
         SimpleDateFormat requiredFormat = new SimpleDateFormat("yyyy-MM-dd");
+        if (beginDate.charAt(4) != '-') { // for two patterns MM-dd-yyyy & yyyy-MM-dd
```

Comment:
please format properly, and more than this what happens in case of the date in is the 'yyyy-MM-dd' format? we still need to validate is a **valid date** (eg. I want to make sure I do not endup passing 2024-29-23' to db

Reply:
Ok. There is a solution